test/GameLogic/GameSession.test.ts

```diff
@@ -110,4 +110,62 @@ describe('GameLogic.GameSession', () => {
 			});
 		});
 	});
+
+	describe('Recordings', () => {
+		it('Should return empty recordings when none added', () => {
+			const session = new GameSession(() => new Level(levelConfig));
+
+			assert.lengthOf(session.getRecordings(), 0);
+		});
+
+		it('Should return empty recordings when none added', () => {
+			const session = new GameSession(() => new Level(levelConfig));
+
+			assert.lengthOf(session.getRecordings(), 0);
+		});
+
+		it('Should return recordings added in constructor', () => {
+			const actionSequences = [
+				new ActionSequence(),
+				new ActionSequence(),
+			];
+			const session = new GameSession(() => new Level(levelConfig), {recordings: actionSequences});
```

Comment:
I'd feel more comfortable is we created an actual GameSessionConfig class and passed that in, rather than an anonymous object that just so happens to be equivalent at this moment in time.

Reply:
What I like about TypeScript is that for simple data holders without any logic you can just create an interface and [duck typing](https://www.typescriptlang.org/docs/handbook/interfaces.html) will ensure the structure is correct on compilation.

I don't particularly mind rewriting it to be a class, I just don't think it will bring any value. It's checked on compilation time and any IDE should also be able to figure out auto completion. It's also a pretty common pattern (using interfaces for data-holders in TS).

If you're fine with keeping it as-is to see if you get more comfortable with this pattern I'd rather keep it. We can always change it to a class later.